posthog/src/main/java/com/posthog/PostHog.kt

```diff
@@ -466,6 +469,35 @@ public class PostHog private constructor(
         }
     }
 
+    /**
+     * Because we use UUIDv7
+     * we can convert a session id into the session start timestamp
+     * and then use that to enforce the maximum session length
+     */
+    private fun enforceMaxSessionLength(buildProperties: Map<String, Any>): Map<String, Any> {
+        val sessionId = buildProperties["\$session_id"] ?: return buildProperties
+
+        val sessionStartTimestamp =
+            TimeBasedEpochGenerator.getTimestampFromUuid(UUID.fromString(sessionId.toString()))
+                ?: return buildProperties
+
+        val currentTimeMillis =
+            config?.dateProvider?.currentTimeMillis() ?: System.currentTimeMillis()
+        val sessionAge = currentTimeMillis - sessionStartTimestamp
+        if (sessionAge > MAX_SESSION_AGE_MILLIS) {
+            // rotate session
+            // update properties
+            PostHogSessionManager.endSession()
+            PostHogSessionManager.startSession(currentTimeMillis)
```

Comment:
this changes the session id but the session replay would need to be notified to eg send a new full snapshot with metadata etc otherwise it'll assume its already sent and the replay will be broken (missing meta event and full snapshot for the new session id)

Reply:
🤔 in the JS SDK we have an event emitter that we pass around that can be used for this kind of "onBlah" behavior

I guess the sessionManager can have a `onSessionStarted` that accepts a callback?

Is there a more android-y way of doing it?

Reply:
yeah a callback would be one way, since `PostHogSessionManager` lives in the java-only lib and most of the replay code is Android-dependent, we will need to pass callbacks around, or implement a pub/sub approach for "app is in background, app is in foreground, etc" so the java bits only can do things